app/main.py

```diff
@@ -2,0 +1,58 @@
+class SoftwareEngineer:
+    skills = []
+
+    def __init__(self, name: str) -> None:
+        self.name = name
+        self.skills = []
+
+    def learn_skill(self, skill: str) -> None:
+        if skill not in self.skills:
+            self.skills.append(skill)
+
+
+class FrontendDeveloper(SoftwareEngineer):
+    skills = ["JavaScript", "HTML", "CSS"]
+
+    def __init__(self, name: str) -> None:
+        super().__init__(name)
+        self.skills = FrontendDeveloper.skills.copy()
+
+    def create_awesome_web_page(self) -> str:
+        print(f"{self.name} is creating a webpage...")
+        return "<h1>Hello world</h1>"
+
+
+class BackendDeveloper(SoftwareEngineer):
+    skills = ["Python", "SQL", "Django"]
+
+    def __init__(self, name: str) -> None:
+        super().__init__(name)
+        self.skills = BackendDeveloper.skills.copy()
+
+    def create_powerful_api(self) -> str:
+        print(f"{self.name} is creating an API...")
+        return "http://127.0.0.1:8000"
+
+
+class AndroidDeveloper(SoftwareEngineer):
+    skills = ["Java", "Android studio"]
+
+    def __init__(self, name: str) -> None:
+        super().__init__(name)
+        self.skills = AndroidDeveloper.skills.copy()
```

Comment:
Again, using `AndroidDeveloper.skills.copy()` is appropriate for instance-specific skills. The class variable `skills` in `SoftwareEngineer` should be removed to prevent shared state issues.